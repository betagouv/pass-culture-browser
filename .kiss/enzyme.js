import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
