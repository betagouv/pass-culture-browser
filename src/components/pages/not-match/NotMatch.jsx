import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import { Redirect } from 'react-router-dom'

class NoMatch extends PureComponent {
  constructor(props) {
    super(props)
    this.timer = null
    this.state = { timing: props.delay }
  }

  componentDidMount() {
    const timeout = 1000
    this.timer = setInterval(() => {
      this.setState(({ timing }) => ({ timing: timing - 1 }))
    }, timeout)
  }

  componentWillUnmount() {
    if (this.timer) clearInterval(this.timer)
    this.timer = null
  }

  redirectTo = () => {
    const { redirect } = this.props
    return <Redirect to={redirect} />
  }

  render() {
    const { timing } = this.state
    const { location } = this.props
    if (timing < 0) return this.redirectTo()
    return (
      <div id="page-redirect">
        <h3 className="title">
          {`404 Not found ${location.pathname}`}
        </h3>
        <p className="content">
          {timing > 0 && (
            <span className="is-block">
              {`Vous allez être automatiquement redirigé dans ${timing} secondes`}
            </span>
          )}
          {
            timing === 0 &&
            <span className="is-block">
              {'Redirecting...'}
            </span>
          }
        </p>
      </div>
    )
  }
}

NoMatch.defaultProps = {
  delay: 5,
  redirect: '/',
}

NoMatch.propTypes = {
  delay: PropTypes.number,
  location: PropTypes.shape().isRequired,
  redirect: PropTypes.string,
}

export default NoMatch
