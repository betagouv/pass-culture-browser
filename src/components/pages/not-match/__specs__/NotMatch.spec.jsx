import { mount } from 'enzyme'
import React from 'react'

import { MemoryRouter } from 'react-router'
import NotMatch from '../NotMatch'

describe('notMatch', () => {
  it('should display a message notifying the user they are on a wrong path and add a link to home', () => {
    // when
    const wrapper = mount(
      <MemoryRouter>
        <NotMatch />
      </MemoryRouter>
    )

    // then
    const title = wrapper.find({ children: 'Oh non !' })
    const subtitle = wrapper.find({ children: "Cette page n'existe pas." })
    const redirectionLink = wrapper.find('a[href="/"]')

    expect(title).toHaveLength(1)
    expect(subtitle).toHaveLength(1)
    expect(redirectionLink).toHaveLength(1)
  })
})
