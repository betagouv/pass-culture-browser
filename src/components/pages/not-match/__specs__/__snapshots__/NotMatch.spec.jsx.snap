// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | NotMatch should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotMatch
    delay={1}
    location={Object {}}
    redirect="/fake-url"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="title"
        >
          404 Not found undefined
        </h3>,
        <p
          className="content"
        >
          <span
            className="is-block"
          >
            Vous allez être automatiquement redirigé dans 1 secondes
          </span>
        </p>,
      ],
      "id": "page-redirect",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "404 Not found undefined",
          "className": "title",
        },
        "ref": null,
        "rendered": "404 Not found undefined",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="is-block"
            >
              Vous allez être automatiquement redirigé dans 1 secondes
            </span>,
            false,
          ],
          "className": "content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vous allez être automatiquement redirigé dans 1 secondes",
              "className": "is-block",
            },
            "ref": null,
            "rendered": "Vous allez être automatiquement redirigé dans 1 secondes",
            "type": "span",
          },
          false,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="title"
          >
            404 Not found undefined
          </h3>,
          <p
            className="content"
          >
            <span
              className="is-block"
            >
              Vous allez être automatiquement redirigé dans 1 secondes
            </span>
          </p>,
        ],
        "id": "page-redirect",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "404 Not found undefined",
            "className": "title",
          },
          "ref": null,
          "rendered": "404 Not found undefined",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="is-block"
              >
                Vous allez être automatiquement redirigé dans 1 secondes
              </span>,
              false,
            ],
            "className": "content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Vous allez être automatiquement redirigé dans 1 secondes",
                "className": "is-block",
              },
              "ref": null,
              "rendered": "Vous allez être automatiquement redirigé dans 1 secondes",
              "type": "span",
            },
            false,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
